### PR TITLE
Drop Nuls support

### PR DESCRIPTION
**Pull request by Rgascoin**
The NULS protocol is no longer supported by Aleph services, there is no point to keep it in the SDK.
(NULS 2 is still remaining)

Issue https://github.com/aleph-im/aleph-sdk-ts/issues/39
/!\ This PR include breaking changes

### CODE REVIEW

**Review by BjrInt — APPROVED** [review]

File: src/accounts/nuls2.ts [hunk 1]
```diff
@@ -132,10 +139,93 @@ export async function ImportAccountFromPrivateKey(
...
+/**
+ * Creates a hash from a message.
+ *
+ * @param message The message used to create the hash.
+ * @param messagePrefix The optional message's hash prefix.
+ */
+export function magicHash(message: Buffer, messagePrefix?: string | Buffer): Buffer {
+    if (!messagePrefix) messagePrefix = "\u0018NULS Signed Message:\n";
+    if (!Buffer.isBuffer(messagePrefix)) messagePrefix = Buffer.from(messagePrefix);
+
```
Comment (BjrInt) [inline src/accounts/nuls2.ts:185]: Can you use [default paramaters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters) instead ? It is more explicit.

File: src/accounts/nuls2.ts [hunk 2]
```diff
@@ -132,10 +139,93 @@ export async function ImportAccountFromPrivateKey(
...
+    const sha = new shajs.sha256().update(publicKey).digest();
+    const publicKeyHash = new RIPEMD160().update(sha).digest();
+    const output = Buffer.allocUnsafe(3);
+
+    output.writeInt16LE(chain_id, 0);
+    output.writeInt8(address_type, 2);
+    return Buffer.concat([output, publicKeyHash]);
+}
+
+export function writeVarInt(value: number, buf: Buffer, cursor: number): number {
```
Comment (BjrInt) [inline src/accounts/nuls2.ts:214]: This function has no doc comment.
(I think we could get rid of this function that mutates an unsafe buffer, by rewriting the `magicHash` helper)

File: src/accounts/nuls2.ts [hunk 3]
```diff
@@ -132,10 +139,93 @@ export async function ImportAccountFromPrivateKey(
...
 
     const hash = publicKeyToHash(pub, { chain_id: chain_id });
     const address = addressFromHash(hash, prefix);
     return new NULS2Account(address, publicKey, privateKey);
 }
+
+/********************************
+ * NULS2 TOOLS METHODS
+ ********************************/
+
```
Comment (BjrInt) [inline src/accounts/nuls2.ts:153]: I have mixed feeling toward those tools. 
I get we should provide helpers on the NULS2 chain, but all the other accounts only exports methods to instantiate the Account class. 

I propose we either provide an `utils` file alongside the accounts, with helpers and typedefs, or use those functions as static methods on the NULSAccount class. What do you think ?

**Review by BjrInt — CHANGES_REQUESTED** [review]
Thank you for taking care of this! :+1: 
I've left a few comments which should not drastically impact what you already submitted.

File: src/accounts/nuls2.ts (hunk 2 above)
Comment (Rgascoin) [reply src/accounts/nuls2.ts]: I merged writeVarInt() and magicHash(), see: https://github.com/aleph-im/aleph-sdk-ts/pull/43/commits/e20d579ea6f8e641b4398c0bd720f6538dba3b17

Also, I thought it was not so useful to keep getXor as a tool function without Nuls1 in this context, so I removed it. What did you think about it?

File: src/accounts/nuls2.ts (hunk 3 above)
Comment (Rgascoin) [reply src/accounts/nuls2.ts]: Changed made in: https://github.com/aleph-im/aleph-sdk-ts/pull/43/commits/e20d579ea6f8e641b4398c0bd720f6538dba3b17

As NULS2 is the only chain that needs those tooling functions, making them static in the NULS2Account class makes more sense for me even if it makes the class slightly more blurry.

**Review by BjrInt — APPROVED** [review]

**Comment by BjrInt** [comment]
Seems nice! :ok_hand:

**Pull request merged**